### DISCUSSION TITLE
Where can I find values available to the stylesheet?

### DISCUSSION THREAD

**Discussion by gwennlbh** [gibatronic/ngx-superbindex#49] [Q&A] [answered]
The title says it all. I want to customize superbindex a bit, but I can't find any resource on what values (e.g. I know that the mtime attribute is available to sort on) can be used.

For example, here's what I want to do:

- Add a "go up" arrow that links to the parent folder
- Show dotfiles
- Change the `<title>` to use the current path (`ewen-lbh's files: CURRENT_PATH_HERE`)
- Maybe add a `<h1>` and some explanation `<p>`s to the root directory listing
- Maybe add a small square thumbnail for image files
- Maybe hide symbolinc links
- Suffix directories with a white `/`
- Suffix links with a white `@`
- Display links in a different sections (just like how directories are separated from files)

**Comment by gibatronic** [comment, marked as answer]
Hi there!

To check the underlying XML data, comment out the `xslt_stylesheet` configuration option from your `nginx.conf`.

About changes in `HTML`, the file [client/templates/main.xslt](https://github.com/gibatronic/ngx-superbindex/blob/master/client/templates/main.xslt) is where you do them.
Please, check [CONTRIBUTING.md](https://github.com/gibatronic/ngx-superbindex/blob/master/CONTRIBUTING.md) to see how to set up the project and generate `public/superbindex.xslt`

About dotfiles, [ngx_http_autoindex_module](http://nginx.org/en/docs/http/ngx_http_autoindex_module.html) has no configuration option to show them.

The path in the title is a tricky one, again the [ngx_http_autoindex_module](http://nginx.org/en/docs/http/ngx_http_autoindex_module.html) doesn't provide the current path.
It might be possible to deliver the requested url through [`xslt_string_param`](http://nginx.org/en/docs/http/ngx_http_xslt_module.html#xslt_string_param). But so far I never tried. And if you can, you'd have to strip the origin part out using nothing but `xsl:template`, which I've promised to myself to never fall into that trap again.
I'm afraid you'd have to use javascript for it.

A thumbnail can be done by requesting the file itself on a `<img>` tag. But a bit of effort will have to be made to detect the file extension and show the thumbnail if it ends in `.jpg`, `.png`, etc…
It can be done either in the XSLT or throught javascript.

For the rest, it's pretty much the same effort of working in the XSLT part or throught javascript.

**Reply by gwennlbh** [reply]
Thanks for the detailed answer

I duckduckgoed a bit for showing dotfiles and I stumbled upon a sad truth: [a 7-years-old patch for nginx was proposed to add a `autoindex_show_hidden_files`](https://trac.nginx.org/nginx/attachment/ticket/557/ngx_autoindex_show_hidden_files.patch) but was evidently never merged... The even sadder part is that the code change is pretty trivial... I'm kind of tempted to hack on nginx's source itself and recompile it myself... Which might be a _tad_ overkill haha

> To check the underlying XML data, comment out the xslt_stylesheet configuration option from your nginx.conf.

Thanks! I was kind of disappointed in how little data is available... I was expecting a large and diverse XML tree, but it's just a

```xml
<list>
  <directory or file mtime="in ISO 8061" size="presumably in bytes">filename</directory or file>
</list>
```

**Reply by gibatronic** [reply]
Indeed, the autoindex module is meant to be simple.
I think it'd be a fun challenge to apply that patch and recompile nginx.
